scalatags/shared/src/main/scala/scalatags/generic/Util.scala

```diff
@@ -43,12 +43,13 @@ trait Util[Builder, Output <: FragT, FragT] extends LowPriUtil[Builder, Output,
     /**
      * Converts the string to a [[Attr]]
      */
-    def attr = {
+    def attr: Attr = attr(None)
+    def attr(namespace: Option[Namespace]): Attr = {
```

Comment:
I would investigate whether it's actually necessary to keep an `Option` here. If you look at `JsDom.TypedTag` it *always* uses a namespace and creates tags using `createElementNS`. Maybe it will be totally fine if we do the same thing for attributes, i.e. always call `setAttributeNS`. This way the signature of `attr` would be analogous to `tag`:

```scala
def tag[T <: Output](implicit namespaceConfig: Namespace)
def attr(implicit namespaceConfig: Namespace)
```

The default namespace would be `Namespace.htmlNamespaceConfig` and creation of attributes for different namespaces would be done by importing some other namespace as implicit, like it's done in `scalatags.jsdom.SvgTags`

This way we would keep the API consistent.